index.html

```diff
@@ -107,11 +112,69 @@ <h3><a href="https://github.com/FIRSTMap/firstmap.github.io">View or edit FIRSTM
             </div>
         </div>
 
+        <div id="search">
+            <h2 class="stripe">Search</h2>
+
+            <div id="searchSection" hidden>
+                <h3>Search</h3>
+                <div id="teamSearch">
+                    <input type="search" id="teamSearchInput" onkeyup="inputKeyUp(this)" />
+                    <button onclick="search()">Search</button>
+                </div>
+
+                <div id="eventSearch" hidden>
+                    <input type="search" id="eventSearchInput" />
+                    
+                    <ul id="eventSearchList">
+                    </ul>
+                </div>
+
+                <select id="searchType" onchange="(() => { updateSearchType(); })()">
+                    <option value="team">Team</option>
+                    <option value="event">Event</option>
+                </select>
+                <br />
+                <h6 id="searchInfoText"></h6>
+            </div>
+
+            <div id="filterSection" hidden>
+                <h3>Filter</h3>
+                <div id="teamFilter">
+                    <input type="search" id="filterInputTeams" onkeyup="inputKeyUp(this)" />
+                    <button onclick="filter()">Filter</button>
+                </div>
+
+                <div id="eventDistrictFilter" hidden>
+                    <input type="search" id="filterInputOther" />
+                    
+                    <ul id="filterInputList">
+                    </ul>
+                </div>
+                
+                <select id="filterType" onchange="(() => { updateFilterType(); clearFilterClick(); })()">
```

Comment:
Naming a function through `onclick` is one thing but defining a function in the HTML code is especially bad practice